### PR TITLE
Fluent: polish copilot variant

### PR DESCRIPTION
**Pull request by OEvgeny**
> Fixes #

## Changelog Entry

- Enhanced the visual presentation of the Fluent theme copilot variant, in PR [#5293](https://github.com/microsoft/BotFramework-WebChat/pull/5293), by [@OEvgeny](https://github.com/OEvgeny)

## Description

This PR refines the appearance of AI-generated messages in the Fluent theme `copilot` variant, focusing on improving the visual consistency and user experience. The changes include enhancements to the copilot message header, activity decoration, and avatar handling.

## Design

We now use author from message entity for bot title and avatar while falling back to either activity or global `styleOptions`.

## Specific Changes

- Updated `CopilotMessageHeader` component to use dynamic avatar images and author information
- Created new utility functions: `isAIGeneratedActivity`, `useActivityAuthor`, and `useActivityStyleOptions`
- Refined CSS styles in `ActivityDecorator.module.css` and `CopilotMessageHeader.module.css`
- Updated `PreChatMessageActivity` to use the new `useActivityAuthor` hook
- Modified theme styles in `Theme.module.css` for improved focus states and borders
- Updated SendBox to show exceeded message length only when exceeded
- Fixed Flair animation is running multiple times when show/hide
- Updated test files to include new scenarios for AI-generated messages and streaming

## -

-  [x] I have added tests and executed them locally
-  [x] I have updated `CHANGELOG.md`
-  [ ] ~I have updated documentation## Review Checklist~

> This section is for contributors to review your work.

-  [ ] Accessibility reviewed (tab order, content readability, alt text, color contrast)
-  [ ] Browser and platform compatibilities reviewed
-  [ ] CSS styles reviewed (minimal rules, no `z-index`)
-  [ ] Documents reviewed (docs, samples, live demo)
-  [ ] Internationalization reviewed (strings, unit formatting)
-  [ ] `package.json` and `package-lock.json` reviewed
-  [ ] Security reviewed (no data URIs, check for nonce leak)
-  [ ] Tests reviewed (coverage, legitimacy)

### CODE REVIEW

File: __tests__/__image_snapshots__/html/side-by-side-wide-dark-js-fluent-theme-applied-dark-theme-applied-side-by-side-left-transcript-right-pre-chat-message-6-snap.png
Comment (compulim) [inline __tests__/__image_snapshots__/html/side-by-side-wide-dark-js-fluent-theme-applied-dark-theme-applied-side-by-side-left-transcript-right-pre-chat-message-6-snap.png:1]: The new screenshot seems to made the highlight ring a bit worse.

File: __tests__/html/fluentTheme/side-by-side.wide.dark.html [hunk 1]
```diff
@@ -438,10 +470,49 @@
...
+          from: { role: 'channel' },
           text: `Extension refreshed`,
           type: 'message'
         }
+      ], [
+        {
+          timestamp: timestamp(),
+          from: { role: 'bot' },
+          entities: [aiMessageEntity],
+          "id": "a4c0c01d-c06e-4dde-9278-265c607b545b",
```
Comment (compulim) [inline __tests__/html/fluentTheme/side-by-side.wide.dark.html:482]: Curious why some of the field names are double-quoted. 🤣

File: packages/fluent-theme/src/components/activity/CopilotMessageHeader.module.css [hunk 2]
```diff
@@ -38,5 +39,6 @@
   font-size: var(--webchat-fontSizeBase100);
   height: 20px;
   line-height: var(--webchat-lineHeightBase100);
-  padding-inline: 4px;
+  padding-inline: var(--webchat-spacingHorizontalXS);
+  flex: none;
```
Comment (compulim) [inline packages/fluent-theme/src/components/activity/CopilotMessageHeader.module.css:43]: Sort.

File: packages/fluent-theme/src/components/activity/CopilotMessageHeader.tsx [hunk 3]
```diff
@@ -1,31 +1,39 @@
...
+
   return (
     <div className={classNames['copilot-message-header']}>
-      <img
-        alt={localize('AVATAR_ALT', botTitle)}
-        className={classNames['copilot-message-header__avatar']}
-        src={botAvatarImage}
-        style={avatarStyle}
-      />
+      {avatarImage && (
```
Comment (compulim) [inline packages/fluent-theme/src/components/activity/CopilotMessageHeader.tsx:29]: Do we have a test to cover this case?

File: __tests__/__image_snapshots__/html/side-by-side-wide-dark-js-fluent-theme-applied-dark-theme-applied-side-by-side-left-transcript-right-pre-chat-message-6-snap.png
Comment (OEvgeny) [reply __tests__/__image_snapshots__/html/side-by-side-wide-dark-js-fluent-theme-applied-dark-theme-applied-side-by-side-left-transcript-right-pre-chat-message-6-snap.png:1]: Fixed. Looks like it got cut :rofl:

File: packages/fluent-theme/src/components/activity/CopilotMessageHeader.module.css (hunk 2 above)
Comment (OEvgeny) [reply packages/fluent-theme/src/components/activity/CopilotMessageHeader.module.css:43]: Both should work. I prefer flex: auto/none

File: packages/fluent-theme/src/components/activity/private/useActivityAuthor.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { useMemo } from 'react';
+import { getOrgSchemaMessage, type WebChatActivity } from 'botframework-webchat-core';
+
+export default function useActivityAuthor(activity?: WebChatActivity | undefined) {
+  return useMemo(() => {
```
Comment (compulim) [inline packages/fluent-theme/src/components/activity/private/useActivityAuthor.ts:5]: Maybe typing it here so we are sure the structure in line 8-13 is a `Person`. 

```ts
  return useMemo<Person>(() => {
```

File: packages/fluent-theme/src/components/activity/private/useActivityAuthor.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    const entity = getOrgSchemaMessage(activity?.entities || []);
+    return typeof entity?.author === 'string'
+      ? {
+          '@type': 'Person',
+          description: undefined,
+          image: undefined,
```
Comment (compulim) [inline packages/fluent-theme/src/components/activity/private/useActivityAuthor.ts:11]: Do we need to specify them as `undefined`?

File: packages/fluent-theme/src/components/decorator/private/BorderFlair.tsx [hunk 6]
```diff
@@ -1,15 +1,28 @@
...
 import { useStyles } from '../../../styles';
 import styles from './BorderFlair.module.css';
 
 function BorderFlair({ children }: Readonly<{ children?: ReactNode | undefined }>) {
   const classNames = useStyles(styles);
+  const [isComplete, setComplete] = useState(false);
+
+  const handleAnimationEnd = useCallback(
+    event =>
+      (event.animationName === styles['borderAnimation-angle'] ||
```
Comment (compulim) [inline packages/fluent-theme/src/components/decorator/private/BorderFlair.tsx:13]: Ah, I should have done this to my PR.

File: packages/fluent-theme/src/components/sendBox/SendBox.module.css [hunk 7]
```diff
@@ -10,6 +10,13 @@
   --webchat-sendbox-border-radius: var(--webchat-borderRadiusLarge);
 }
 
+/* Copilot variant */
+:global(.webchat-fluent) .sendbox.variant-copilot {
```
Comment (compulim) [inline packages/fluent-theme/src/components/sendBox/SendBox.module.css:14]: Do we have VRT for this?

**Review by compulim — DISMISSED** [review]
Nitpicking.

File: packages/fluent-theme/src/components/activity/CopilotMessageHeader.tsx (hunk 3 above)
Comment (OEvgeny) [reply packages/fluent-theme/src/components/activity/CopilotMessageHeader.tsx:29]: No test for this one

File: packages/fluent-theme/src/components/activity/private/useActivityAuthor.ts (hunk 4 above)
Comment (OEvgeny) [reply packages/fluent-theme/src/components/activity/private/useActivityAuthor.ts:5]: The type is not shared, so we don't rely on it.

File: packages/fluent-theme/src/components/activity/private/useActivityAuthor.ts (hunk 5 above)
Comment (OEvgeny) [reply packages/fluent-theme/src/components/activity/private/useActivityAuthor.ts:11]: Needed initially to convey typescript that this is `Person` compatible

File: packages/fluent-theme/src/components/sendBox/SendBox.module.css (hunk 7 above)
Comment (OEvgeny) [reply packages/fluent-theme/src/components/sendBox/SendBox.module.css:14]: Not for the case when it overflows at least.

**Review by compulim — APPROVED** [review]

**Pull request merged**